File: api/app/schemas/vehicle.py
from typing import Optional
from uuid import UUID
# pyrefly: ignore [missing-import]
from pydantic import EmailStr
from app.schemas.common import ORMModel


class VehicleBase(ORMModel):
    owner_name: str
    owner_phone: Optional[str] = None
    owner_email: Optional[EmailStr] = None
    brand: str
    model: str
    year: Optional[int] = None
    plate: Optional[str] = None
    vin: Optional[str] = None
    mileage: Optional[int] = None


class VehicleCreate(VehicleBase):
    pass


class VehicleUpdate(VehicleBase):
    pass


class VehicleOut(VehicleBase):
    id: UUID

File: client/src/features/Home/models/Vehicle.ts
import { EstimateItemCreate, EstimateStatus } from "./Estimate";

export type Row = {
  id?: string;
  estimate_id?: string;
  owner_name: string;
  owner_phone?: string | null;
  owner_email?: string | null;
  brand: string;
  model: string;
  year?: number | null;
  plate?: string | null;
  vin?: string | null;
  mileage?: number | null;
  customer_notes?: string | null;
  internal_notes?: string | null;
  tax_percent?: number | null;
  discount_amount?: number | null;
  validity_days?: number | null;
  subtotal?: number;
  tax_amount?: number;
  total?: number;
  status?: EstimateStatus | undefined;
  items?: EstimateItemCreate[];
};

export interface VehicleCreate {
  owner_name: string;
  owner_phone?: string | null;
  owner_email?: string | null;
  brand: string;
  model: string;
  year?: number | null;
  plate?: string | null;
  vin?: string | null;
  mileage?: number | null;
}

export interface VehicleOut extends VehicleCreate {
  id: string;
}
